operators/pkg/controller/elasticsearch/certificates/http/reconcile.go

```diff
@@ -113,6 +107,51 @@ func reconcileHTTPInternalCertificateSecret(
 		secret.Data = make(map[string][]byte)
 	}
 
+	if customCertificates != nil {
+		if !reflect.DeepEqual(secret.Data, customCertificates.Data) {
+			needsUpdate = true
+			secret.Data = customCertificates.Data
+		}
+	} else {
+		selfSignedNeedsUpdate, err := verifyInternalSelfSignedCertificateSecret(
+			&secret, es, svcs, ca, certValidity, certReconcileBefore,
+		)
+		if err != nil {
+			return nil, err
+		}
+		needsUpdate = needsUpdate || selfSignedNeedsUpdate
+	}
+
+	if needsUpdate {
+		if shouldCreateSecret {
+			log.Info("Creating HTTP internal certificate secret", "secret", secret.Name)
+			if err := c.Create(&secret); err != nil {
+				return nil, err
+			}
+		} else {
+			log.Info("Updating HTTP internal certificate secret", "secret", secret.Name)
+			if err := c.Update(&secret); err != nil {
+				return nil, err
+			}
+		}
+	}
+
```

Comment:
the if/else flow of this function is rather complex, but I don't see an obvious way of improving it :/

Reply:
Could refactor or create a new generic reconcile resource method, because this is not the only place something like this shows up. Out of scope for this PR imo though